src/main/python/fine_tuning/x_fold_cv.py

```diff
@@ -0,0 +1,121 @@
+                    for i in xrange(self.fold):
+                        test_fold_performances = avg_performances[basemodel][model][metric][i]
+                        training_data = {}
+                        for j in xrange(self.fold):
+                            if j == i:
+                                continue
+                            fold_performance = avg_performances[basemodel][model][metric][j]
+                            for param in  fold_performance:
+                                if param not in training_data:
+                                    training_data[param] = .0
+                                training_data[param] += fold_performance[param]
+                        sorted_training_performance = sorted(training_data.items(),
+                                                             key=lambda x:x[1],
+                                                             reverse=True)
+                        best_param = sorted_training_performance[0][0]
+                        if not quite:
+                            print "\tFold: %d" %(i)
+                            print "\t\tBest param: %s" %(best_param)
+                            print "\t\ttest performance: {0:.4f}".format(test_fold_performances[best_param])
+
+                        metric_fold_performances.append(test_fold_performances[best_param])
+                    print "\tAverage {0:.4f}".format( sum(metric_fold_performances) / len(metric_fold_performances))
+
+    def _get_param_avg_performances(self,file_path):
+        param_performance_list = {}
+        for i in xrange(self.fold):
+            param_performance_list[i] = {}
+        with open(file_path) as f:
+            for line in f:
+                line = line.strip()
+                if line:
+                    row = line.split()
+                    metric = row[0]
+                    if metric not in param_performance_list[0]:
+                        for i in param_performance_list:
+                            param_performance_list[i][metric] = []
+                    qid = row[1]
+                    try:
+                        value = float(row[2])
+                    except:
+                        self.logger.error( "Cannot parse %s" %(row[2]) )
+                        continue
+                    else:
+                        if qid != "all":
+                            # compute fold id base on qid
+                            idx = int(qid) % self.fold
+                            param_performance_list[idx][metric].append(value)
+        param_avg_performances = {}
+
+        for metric in param_performance_list[0].keys():
+            param_avg_performances[metric] = {}
+            for i in param_performance_list:
+                param_avg_performances[metric][i] = sum(param_performance_list[i][metric])/len(param_performance_list[i][metric])
+        return param_avg_performances
+
+def main():
+    parser = argparse.ArgumentParser(description=__doc__)
```

Comment:
We should add an arg option to validate the results, i.e. to put the results we finally get and compare the results with those.

Reply:
So you mean we use the `expected` value in `model.yaml` to do that?

Reply:
Actually, I think we should only expose one interface, i.e. `run_batch.py`
That interface streamlines everything.
But we can do that later

Reply:
Sure.